cpu/esp32/periph/dac.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_esp32
+ * @ingroup     drivers_periph_adc
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "periph/dac.h"
+
+#include "adc_arch.h"
+#include "adc_ctrl.h"
+#include "esp_common.h"
+#include "gpio_arch.h"
+#include "rom/ets_sys.h"
+#include "soc/rtc_io_struct.h"
+#include "soc/rtc_cntl_struct.h"
+#include "soc/sens_reg.h"
+#include "soc/sens_struct.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/** Map of RIOT DAC lines to GPIOs */
+static const uint32_t dac_pins[] = DAC_GPIOS;
+
+/** number of DAC channels */
+const unsigned dac_chn_num = (sizeof(dac_pins) / sizeof(dac_pins[0]));
+
+/* declaration of external functions */
+extern void _adc2_ctrl_init(void);
+
+/* forward declarations of internal functions */
+static bool _dac_conf_check(void);
+static bool _dac_module_initialized  = false;
+
+/* external variable declarations */
+extern const gpio_t _gpio_rtcio_map[];
+
+int8_t dac_init (dac_t line)
+{
+    CHECK_PARAM_RET (line < dac_chn_num, DAC_NOLINE)
+
+    if (!_dac_module_initialized) {
+        /* do some configuration checks */
+        if (!_dac_conf_check()) {
+            return -1;
```

Comment:
Is it possible to use the error code `DAC_NOLINE` here?

Reply:
Yes